pkg/controller/egressgateway/egressgateway_controller.go

```diff
@@ -0,0 +1,666 @@
+			}
+			return reconcile.Result{}, nil
+		}
+		reqLogger.Error(err, "Error querying installation")
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error querying installation", err, reqLogger)
+		for _, egw := range egwsToReconcile {
+			r.setDegraded(ctx, &egw, "Error querying installation", err)
+		}
+		return reconcile.Result{}, err
+	}
+
+	variant := installStatus.Variant
+	if variant != operatorv1.TigeraSecureEnterprise {
+		reqLogger.Error(err, fmt.Sprintf("Waiting for network to be %s", operatorv1.TigeraSecureEnterprise))
+		r.status.SetDegraded(operatorv1.ResourceNotReady, fmt.Sprintf("Waiting for network to be %s", operatorv1.TigeraSecureEnterprise), nil, reqLogger)
+		for _, egw := range egwsToReconcile {
+			r.setDegraded(ctx, &egw, fmt.Sprintf("Waiting for network to be %s", operatorv1.TigeraSecureEnterprise), nil)
+		}
+		return reconcile.Result{}, nil
+	}
+
+	if installStatus.CalicoVersion != components.EnterpriseRelease {
+		return reconcile.Result{RequeueAfter: 30 * time.Second}, nil
+	}
+
+	pullSecrets, err := utils.GetNetworkingPullSecrets(installation, r.client)
+	if err != nil {
+		reqLogger.Error(err, "Error retrieving pull secrets")
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error retrieving pull secrets", err, reqLogger)
+		for _, egw := range egwsToReconcile {
+			r.setDegraded(ctx, &egw, "Error retrieving pull secrets", err)
+		}
+		return reconcile.Result{}, err
+	}
+
+	// Fetch any existing default FelixConfiguration object.
+	fc := &crdv1.FelixConfiguration{}
+	err = r.client.Get(ctx, types.NamespacedName{Name: "default"}, fc)
+	if err != nil && !errors.IsNotFound(err) {
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error reading FelixConfiguration", err, reqLogger)
+		for _, egw := range egwsToReconcile {
+			r.setDegraded(ctx, &egw, "Error reading felix configuration", err)
+		}
+		return reconcile.Result{}, err
+	}
+
+	// Reconcile all the EGWs
+	var errMsgs []string
+	for _, egw := range egwsToReconcile {
+		_, err = r.reconcileEgressGateway(ctx, &egw, reqLogger, variant, fc, pullSecrets, installation, unreadyEGW)
+		if err != nil {
+			reqLogger.Error(err, "Error reconciling egress gateway")
+			errMsgs = append(errMsgs, err.Error())
+		}
+	}
+	if len(errMsgs) != 0 {
+		return reconcile.Result{}, fmt.Errorf(strings.Join(errMsgs, ";"))
+	}
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileEgressGateway) setDegraded(ctx context.Context, egw *operatorv1.EgressGateway, msg string, err error) {
+	if err != nil {
+		setDegraded(r.client, ctx, egw, reconcileErr, fmt.Sprintf("%s Name = %s, Namespace = %s, err = %s", msg, egw.Name, egw.Namespace, err.Error()))
+	} else {
+		setDegraded(r.client, ctx, egw, reconcileErr, fmt.Sprintf("%s Name = %s, Namespace = %s", msg, egw.Name, egw.Namespace))
+	}
+}
+
+func (r *ReconcileEgressGateway) reconcileEgressGateway(ctx context.Context, egw *operatorv1.EgressGateway, reqLogger logr.Logger,
```

Comment:
This function has a massive signature. Any idea on how we can reduce the number of required arguments?

Reply:
This function shouldn't be returning a reconcile.Result, as it's not the main reconcile function.

Really, it should return just an error, and then the calling code should handle it IMO

Reply:
In fact, the reconcile.Result appears to be unused where this function is called anyway.